tests/subsys/bluetooth/mesh/sensor_subsys/src/main.c

```diff
@@ -901,6 +901,270 @@ static void temp_check(const struct bt_mesh_sensor_type *sensor_type)
+	 * and unknown value, respectively) 66560640
+	 */
+	int32_t test_vector_time[] = {0, 1, 2999, 66560640, 0xFFFFFFFE,
+				      0xFFFFFFFF};
+	uint8_t expected_time[] = {0, 64, 148, 253, 0xFE, 0xFF};
+	struct __packed {
+		uint16_t expected_voltage;
+		uint8_t expected_time;
+	} expected;
+
+	memset(in_value, 0, sizeof(in_value));
+
+	for (int i = 0; i < ARRAY_SIZE(test_vector_voltage); i++) {
+		in_value[0].val1 = test_vector_voltage[i] > 159 ?
+				UINT16_MAX : test_vector_voltage[i];
+		in_value[0].val2 = 0;
+
+		in_value[1].val1 = test_vector_time[i];
+		in_value[1].val2 = 0;
+
+		expected.expected_voltage = test_vector_voltage[i] > 159 ?
+				UINT16_MAX : raw_scalar_value_get(test_vector_voltage[i], 1, 0, -6);
+		expected.expected_time = expected_time[i];
+
+		encoding_checking_proceed(sensor_type, in_value, &expected, sizeof(expected));
+		decoding_checking_proceed_with_tolerance(sensor_type, &expected,
+				sizeof(expected), in_value, 0.001);
+	}
+}
+
+static void input_current_stat_check(const struct bt_mesh_sensor_type *sensor_type)
+{
+	struct sensor_value in_value[sensor_type->channel_count];
+	uint16_t test_vector_current[] = {0, 1234, 5555, 11111, 65534, UINT16_MAX};
+
+	/* Time exponential, represented = 1.1^(N-64), where N is raw value.
+	 * Expected raw values precomputed. (0, 0xFE and 0xFF map
+	 * to themselves as they represent 0 seconds, total device lifetime
+	 * and unknown value, respectively) 66560640
+	 */
+	int32_t test_vector_time[] = {0, 1, 2999, 66560640, 0xFFFFFFFE,
+				      0xFFFFFFFF};
+	uint8_t expected_time[] = {0, 64, 148, 253, 0xFE, 0xFF};
+	struct __packed {
+		uint16_t expected_current[4];
+		uint8_t expected_time;
+	} expected;
+
+	memset(in_value, 0, sizeof(in_value));
+
+	for (int i = 0; i < ARRAY_SIZE(test_vector_current); i++) {
+		for (int j = 0; j < sensor_type->channel_count - 1; j++) {
+			in_value[j].val1 = test_vector_current[i] == UINT16_MAX ?
+					UINT16_MAX : test_vector_current[i] / 100;
+			in_value[j].val2 = test_vector_current[i] == UINT16_MAX ?
+					0 : test_vector_current[i] % 100 * 10000;
+			expected.expected_current[j] = test_vector_current[i];
+		}
+
+		in_value[sensor_type->channel_count - 1].val1 = test_vector_time[i];
+		in_value[sensor_type->channel_count - 1].val2 = 0;
+		expected.expected_time = expected_time[i];
+
+		encoding_checking_proceed(sensor_type, in_value, &expected, sizeof(expected));
+		decoding_checking_proceed_with_tolerance(sensor_type, &expected,
+				sizeof(expected), in_value, 0.001);
+	}
+}
+
+static void input_voltage_stat_check(const struct bt_mesh_sensor_type *sensor_type)
+{
+	struct sensor_value in_value[sensor_type->channel_count];
+	uint16_t test_vector_voltage[] = {0, 123, 159, 160, UINT16_MAX};
+
+	/* Time exponential, represented = 1.1^(N-64), where N is raw value.
+	 * Expected raw values precomputed. (0, 0xFE and 0xFF map
+	 * to themselves as they represent 0 seconds, total device lifetime
+	 * and unknown value, respectively) 66560640
```

Comment:
This "66560640" should not be here in the comment. I must have missed this in my PR and now it has been copied to all of your checks, so technically my fault, whoops :) It would be nice if you could remove it from the temperature check as well.